src/makeCalculator.js

```diff
@@ -4,7 +4,43 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  const calculator = {
+    result: 0,
+
+    add(number) {
+      this.result += number;
+
+      return this;
+    },
+    subtract(number) {
+      this.result -= number;
+
+      return this;
+    },
+    multiply(number) {
+      this.result = this.result * number;
+
+      return this;
+    },
+    divide(number) {
+      this.result = this.result / number;
+
+      return this;
+    },
+    reset() {
+      this.result = 0;
+
+      return this;
+    },
+
+    operate(action, number) {
+      action.call(this, number);
```

Comment:
The `operate` method uses `action.call(this, number)`, which assumes that `action` is a function. It would be beneficial to add a check to ensure `action` is a valid function before calling it to prevent runtime errors.